app/controllers/BTAPartialController.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.time.LocalDate
+import javax.inject.{Inject, Singleton}
+
+import config.AppConfig
+import controllers.predicates.AsyncActionPredicate
+import models.{LastTaxCalculation, NoLastTaxCalculation, ObligationModel}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{AnyContent, Action}
+import services.BTAPartialService
+
+@Singleton
+class BTAPartialController @Inject()(implicit val config: AppConfig,
+                                     implicit val messagesApi: MessagesApi,
+                                     val actionPredicate: AsyncActionPredicate,
+                                     val btaPartialService: BTAPartialService
+                                      ) extends BaseController {
+
+  val setupPartial: Int => Action[AnyContent] = taxYear => actionPredicate.async {
```

Comment:
The partial needs to be TaxYear agnostic - BTA will not be able to determine this. So this needs to do some logic to work out which TaxYears are relevant based on their IncomeSources